trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestHelper.java

```diff
@@ -0,0 +1,298 @@
+
+        final StringBuilder stringBuilder = new StringBuilder();
+
+        stringBuilder.append("\n");
+        stringBuilder.append(getTestNameBorderLogLine(lineLength));
+        stringBuilder.append(getTestNameEmptyLogLine(lineLength));
+        stringBuilder.append(getTestNameLogLine(text, paddingLength));
+        stringBuilder.append(getTestNameEmptyLogLine(lineLength));
+        stringBuilder.append(getTestNameBorderLogLine(lineLength));
+
+        logger.lifecycle(stringBuilder.toString());
+    }
+
+    /**
+     * Prints multiple lines with the name of the test in a pretty ASCII text box.
+     *
+     * @param testName the name of the test to print.
+     */
+    public void logTestNameInAsciiBox(@NonNull final TestName testName) {
+        logInAsciiBox(testName.getMethodName());
+    }
+
+    /**
+     * Prints the top or bottom line. Contains separators only.
+     *
+     * @param lineLength the length of the lines.
+     * @return the line.
+     */
+    @NonNull
+    private String getTestNameBorderLogLine(final int lineLength) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("+");
+        stringBuilder.append(concatWord("=", lineLength));
+        stringBuilder.append("+");
+        stringBuilder.append("\n");
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Prints a line with empty content.
+     *
+     * @param lineLength the length of the lines.
+     * @return the line.
+     */
+    @NonNull
+    private String getTestNameEmptyLogLine(final int lineLength) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("|");
+        stringBuilder.append(concatWord(" ", lineLength));
+        stringBuilder.append("|");
+        stringBuilder.append("\n");
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Prints the line with the name of the test.
+     *
+     * @param testName      the given text to display.
+     * @param paddingLength the length of padding for the line, excluding starting and ending cahars.
+     * @return the line.
+     */
+    @NonNull
+    private String getTestNameLogLine(final String testName, final int paddingLength) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("|");
+        stringBuilder.append(concatWord(" ", paddingLength / 2));
+        stringBuilder.append(testName);
+        stringBuilder.append(concatWord(" ", paddingLength / 2));
+        stringBuilder.append("|");
+        stringBuilder.append("\n");
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Concats the given word the given amount of times.
+     *
+     * @param word  the word to concat.
+     * @param times the number of times it should be concatenated.
+     * @return the concatenated String.
+     */
+    @NonNull
+    private String concatWord(@NonNull final String word, final int times) {
```

Comment:
nit: this doesn't appear to be used to concat words but single chars (L123 it's using a space char, L88 it's an equals char). Would it be more appropriate to rename to concatString instead?

Reply:
Sure, we can concat non words too, so concatString would make more sense.

Reply:
thank you :)